src/main/java/htsjdk/samtools/cram/ref/ReferenceSource.java

```diff
@@ -116,7 +110,7 @@ else if (Defaults.USE_CRAM_REF_DOWNLOAD) {
     }
 
     private byte[] findInCache(final String name) {
-        final WeakReference<byte[]> weakReference = cacheW.get(name);
+        final SoftReference<byte[]> weakReference = cache.get(name);
         if (weakReference != null) {
```

Comment:
Can you explicitly check for an empty array here as well, and throw an informative error if you encounter one?

Reply:
Under what circumstances the bases array would be an empty array as supposed to a null (claimed by the GC)?